### PR TITLE
docs about usage of internal services in an "api-gateway" access model

### LINKED ISSUE #3562: How to do extended ingress routing with "httpProtocol: Redirected" setting?
**Describe the change you'd like to see**
I am referring to the documentation in https://knative.dev/docs/serving/samples/knative-routing-go/index.html.

I currently have KNative set up together with cert-manager for automatic certificate provisioning (using the `autoTLS` feature). I would also like to use `"httpProtocol": "Redirected"` to forward all http traffic to https on the istio-ingressgateway.

Doing so make it impossible to follow the suggested solution for routing (described here: https://knative.dev/docs/serving/samples/knative-routing-go/index.html#how-it-works). KNative / istio will internally forward the traffic back to the istio-ingressgateway with a different setting for the `Host` header (by using `rewrite: authority` here: https://github.com/knative/docs/blob/release-0.21/docs/serving/samples/knative-routing-go/routing.yaml#L34). It uses HTTP there, and not HTTPS. I don't necessarily want it to use more than HTTP for the internal stuff.

However, the istio-ingressgateway will try to redirect to HTTPS, making the whole thing fail.

I am willing to make a pull request with documentation updates, but honestly don't really know how one would solve it. If you give me a hint how to do it, I'll make the PR :).

**Additional context**

grafik

### PR DESCRIPTION
**Pull request by lenalebt**
Fixes #3562

## Proposed Changes 

I promised to write down how to support `"httpProtocol": "Redirected"` for the documentation found at https://knative.dev/docs/serving/samples/knative-routing-go/index.html

Thanks to @nak3 I found out how to do it, and wanted to make others lives easier by documenting it.

In case you think it would be better, I can also add graphics - but I did not have the sources of the already-present image, and did not want to make a bad copy of it.

### CODE REVIEW

File: docs/serving/samples/knative-routing-go/index.md
Comment (nak3) [inline docs/serving/samples/knative-routing-go/index.md:266]:
```suggestion
The original ones provided by Knative Serving (`search-service.default.example.com` and `login-service.default.example.com`),
```

I think we use `Knative` instead of `KNative` so it should be consistent.

File: docs/serving/samples/knative-routing-go/sample-internal.yaml
Comment (nak3) [inline docs/serving/samples/knative-routing-go/sample-internal.yaml:20]: I think visibility setting is not `annotations` but `labels` - https://knative.dev/docs/serving/cluster-local-route/

File: docs/serving/samples/knative-routing-go/sample-internal.yaml
Comment (nak3) [inline docs/serving/samples/knative-routing-go/sample-internal.yaml:42]: Same here, it should be `labels`?

File: docs/serving/samples/knative-routing-go/index.md
Comment (nak3) [inline docs/serving/samples/knative-routing-go/index.md:280]: Before applying this yaml, we need to replace the images in `sample-internal.yaml`  by `perl -pi -e "s@github.com/knative/docs/docs/serving/samples@${REPO}@g" docs/serving/samples/knative-routing-go/sample-internal.yaml` again?

I think we should either add `sample-internal.yaml` in `Setup` section above or to run command `kubectl label kservice search-service login-service networking.knative.dev/visibility=cluster-local` instead of `kubectl apply`

**Comment by csantanapr** [comment]
Thanks @nak3 for the review

File: docs/serving/samples/knative-routing-go/sample-internal.yaml
Comment (lenalebt) [reply docs/serving/samples/knative-routing-go/sample-internal.yaml:20]: Absolutely!

File: docs/serving/samples/knative-routing-go/index.md
Comment (lenalebt) [reply docs/serving/samples/knative-routing-go/index.md:280]: I like the `kubectl label kservice search-service login-service networking.knative.dev/visibility=cluster-local` approach. Was not really happy with duplicating the file anyways.

**Comment by lenalebt** [comment]
Thank you for the review! I applied your suggestions, and do really like them better.

File: docs/serving/samples/knative-routing-go/index.md
Comment (nak3) [inline docs/serving/samples/knative-routing-go/index.md:272]:
```suggestion
to the Knative Service definitions, and route traffic over
```
One more `KNative`.

**Review by nak3 — APPROVED** [review]
Thank you so much! Looks good to me except for one more small fix. Let's wait for docs team's another review.

File: docs/serving/samples/knative-routing-go/index.md
Comment (lenalebt) [reply docs/serving/samples/knative-routing-go/index.md]: Sorry, I totally overread this one. Thanks for pointing me to it :)

**Comment by omerbensaadon** [comment]
@lenalebt thanks for your contribution! 

If you're interested in getting more involved, join us in the Knative Slack at https://slack.knative.dev/! Message me and I can help point you in the right direction! :)

**Comment by lenalebt** [comment]
@omerbensaadon Thanks for the nice invitation :). I guess I'll add PRs on-the-go whenever I see something where I do think I can add something. Thanks to @nak3 - it was mostly him pointing me in the right direction here, really thankful for the nice atmosphere!

**Pull request merged**
